From af5b58cb77fc8e2e2e2f26aadef4b773b950eafc Mon Sep 17 00:00:00 2001
From: Chen Qi <Qi.Chen@windriver.com>
Date: Wed, 21 Aug 2024 07:50:57 -0700
Subject: [PATCH] Revert "autotools: make pam install path configurable"

This reverts commit 0539b1ebd26f6a168b822374d7b167f8ef4a1997.

If people want to use /lib instead of /usr/lib as $(libdir) and want
lastlog2 pam module installed in /lib/security, there's no way to do so.

Beside, we have:
pam_lastlog2/meson.build:        pamlibdir = get_option('libdir') / 'security'

So things are not in sync between Makemodule.am and meson.build with
the original commit.

Upstream-Status: Submitted [https://github.com/util-linux/util-linux/pull/3173]

Signed-off-by: Chen Qi <Qi.Chen@windriver.com>
---
 Makefile.am                    | 1 -
 configure.ac                   | 3 ---
 pam_lastlog2/src/Makemodule.am | 3 ++-
 3 files changed, 2 insertions(+), 5 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 54e0fd497..3f801c686 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -56,7 +56,6 @@ usrsbin_exec_PROGRAMS =
 man_MANS =
 pkgconfig_DATA =
 usrlib_exec_LTLIBRARIES =
-usrsecurelib_exec_LTLIBRARIES =
 bin_PROGRAMS =
 sbin_PROGRAMS =
 dist_usrbin_exec_SCRIPTS =
diff --git a/configure.ac b/configure.ac
index 82d0e4bf9..35a70aba4 100644
--- a/configure.ac
+++ b/configure.ac
@@ -120,9 +120,6 @@ AS_CASE([$libdir],
 )
 AC_SUBST([usrlib_execdir])
 
-usrsecurelib_execdir='${usrlib_execdir}/security'
-AC_SUBST([usrsecurelib_execdir])
-
 # static configuration maintaned by packages (e.g. /usr/lib)
 AC_ARG_VAR([SYSCONFSTATICDIR],
 	   [Path to static system configuration, default ${prefix}/lib])
diff --git a/pam_lastlog2/src/Makemodule.am b/pam_lastlog2/src/Makemodule.am
index 6d2365378..f9a9b6b41 100644
--- a/pam_lastlog2/src/Makemodule.am
+++ b/pam_lastlog2/src/Makemodule.am
@@ -1,4 +1,5 @@
-usrsecurelib_exec_LTLIBRARIES += pam_lastlog2.la
+securelibdir = $(libdir)/security
+securelib_LTLIBRARIES = pam_lastlog2.la
 
 pam_lastlog2_la_SOURCES = \
 	pam_lastlog2/src/pam_lastlog2.c
-- 
2.25.1

